kubetest/kops.go

```diff
@@ -43,17 +48,39 @@ var (
 	kopsSSHUser      = flag.String("kops-ssh-user", os.Getenv("USER"), "(kops only) Username for SSH connections to nodes.)")
 	kopsSSHKey       = flag.String("kops-ssh-key", "", "(kops only) Path to ssh key-pair for each node (defaults '~/.ssh/kube_aws_rsa' if unset.)")
 	kopsKubeVersion  = flag.String("kops-kubernetes-version", "", "(kops only) If set, the version of Kubernetes to deploy (can be a URL to a GCS path where the release is stored) (Defaults to kops default, latest stable release.).")
-	kopsZones        = flag.String("kops-zones", "us-west-2a", "(kops only) zones for kops deployment, comma delimited.")
+	kopsZones        = flag.String("kops-zones", "", "(kops only) zones for kops deployment, comma delimited.")
 	kopsNodes        = flag.Int("kops-nodes", 2, "(kops only) Number of nodes to create.")
 	kopsUpTimeout    = flag.Duration("kops-up-timeout", 20*time.Minute, "(kops only) Time limit between 'kops config / kops update' and a response from the Kubernetes API.")
 	kopsAdminAccess  = flag.String("kops-admin-access", "", "(kops only) If set, restrict apiserver access to this CIDR range.")
 	kopsImage        = flag.String("kops-image", "", "(kops only) Image (AMI) for nodes to use. (Defaults to kops default, a Debian image with a custom kubernetes kernel.)")
 	kopsArgs         = flag.String("kops-args", "", "(kops only) Additional space-separated args to pass unvalidated to 'kops create cluster', e.g. '--kops-args=\"--dns private --node-size t2.micro\"'")
 	kopsPriorityPath = flag.String("kops-priority-path", "", "Insert into PATH if set")
-	kopsBaseURL      = flag.String("kops-base-url", "", "Base URL for a prebuilt version of kops")
-	kopsVersion      = flag.String("kops-version", "", "URL to a file containing a valid kops-base-url")
+	kopsBaseURL      = flag.String("kops-base-url", "", "(kops only) Base URL for a prebuilt version of kops")
+	kopsVersion      = flag.String("kops-version", "", "(kops only) URL to a file containing a valid kops-base-url")
 	kopsDiskSize     = flag.Int("kops-disk-size", 48, "Disk size to use for nodes and masters")
 	kopsPublish      = flag.String("kops-publish", "", "Publish kops version to the specified gs:// path on success")
+	kopsMasterType   = flag.String("kops-master-type", "c4.large", "(kops only) master instance type")
```

Comment:
kops-master-size, for consistency with the master-size kops flag

Reply:
I also think default should be empty, because otherwise you're changing the size that all tests run with.

Reply:
Yah I understand that we are changing all tests, and it is a big change.  My overall plan is to always select a random region and at least one random AZ.  Because of that plan, we are going to hit far more regions that do not support m3 instance types.  If we default it to c4.large it may help us with lack of m3 instances. What are your thoughts?